docs/secure-coding-guidelines.md

```diff
@@ -0,0 +1,99 @@
+      2. Limit the amount of time __sensitive__ data is decrypted
+      3. Do not bundle API tokens in client side applications unless you intend for them to be publicly accessible
+   3. Encryption:
+      1. __Sensitive__ data should be secured at rest and in-transit using standard encryption protocols (e.g. cryptography libraries, OS level key management systems, https, etc…)
+      2. Cryptography of __sensitive__ data should meet a minimum standard established by the security team during a threat assessment and annual reviews
+   4. Passwords:
+      1. Passwords used to decrypt content should adhere to security team recommendations during a threat assessment and annual reviews.
+
+3. Logging & Error Handling:
+
+   1. Logging:
+      1. Secrets and __sensitive__ data should not be logged
+      2. Console logging should be removed from production code using linters
+   2. Error Handling:
+      1. All errors should be handled
+      2. Errors should not leak data and be made more generic. This is especially important for code which handles data classified as __sensitive__.
+      3. Production versions of applications shall have logging disabled
+
+4. 3rd Party Integrations & Applications:
+   1. Authentication and Authorization:
+      1. Implement access controls and authentication mechanisms for accessing external data
+      2. Store and manage API keys securely when accessing external sources
+      3. API keys used in client applications shall have a reduced scope adhering to Principle of Least Privilege
+      4. If an API key is not intended to be publicly accessible, do not use it in a client application
+   2. Integration/Application Integrity:
+      1. Monitor integrations/applications for changes and security issues.
+   3. Permissions:
+      1. All applications/dapps/snaps shall adhere to the following to Principle of Least Privilege with a default state of allowing no permissions
+      2. Users shall provide consent for required permissions of applications/dapps/snaps
+
+5. Dependency Management:
+
+   1. Dependency Integrity:
+      1. Use Dependabot to keep all dependencies up to date and to automate checks for updates
+      2. Pin dependency versions or use a yarn.lock file
+   2. Avoid Deprecated and Unmaintained Packages:
+      1. Using Socket.dev to check the health and maintenance status of dependencies and seek alternatives when necessary
+   3. Review Dependency Licenses:
+      1. Ensure dependencies' licenses align with your project's requirements
+   4. Reduce Dependencies and Keep Them Minimal:
+      1. Minimize dependencies to reduce potential vulnerabilities
+   5. Audit and Monitor Dependencies:
+      1. Projects should use the npm audit, Dependabot and Socket.dev tooling for periodic automated auditing of dependencies
+   6. LavaMoat:
+      1. LavaMoat allow-scripts should be enabled on all projects
+      2. LavaMoat runtime should be enabled on all projects
```

Comment:
```suggestion
      2. LavaMoat runtime should be enabled on all projects
        - runtime protections for node.js running your build
        - LavaMoat tools for bundling
      3. When you regenerate the policy file look carefully at the diff. Raise an issue with your team and don't hesitate to call for help in the #lavamoat channel if new items in the policy look like a lot of new privileges or include the ability to **spawn/exec child processes**
      4. Review your policy.json for suspicious permissions at least once in your participation in a project (eg. when you join a new project, familiarize yourself with it and then look at the policy. Ask your team questions as you go.)
```

Reply:
Thanks! I've added something like this here: 62164d7

I'd like to follow up to add some examples of suspicious permissions though. I didn't include the spawn/exec example yet. I want to confirm exactly how they'd show up in the policy first.